### ISSUE TITLE
ICE: translating unsupported cast: `alloc::rc::Rc<[i32]>` in beta

### ISSUE THREAD

**Issue by bluss** [rust-lang/rust#26352]
Current beta release ICE's on trying to create a `Rc<[i32]>`.

Compile error in 1.0, works fine in nightly, so only beta channel.

```rust
use std::rc::Rc;

fn main() {
    let r = Rc::new([1, 2]) as Rc<[i32]>;
}
```

```

error: internal compiler error: translating unsupported cast: alloc::rc::Rc<[i32]> (cast_other) -> alloc::rc::Rc<[i32]> (cast_other)
```

**Comment by bluss** [comment]
cc @nrc

**Comment by nrc** [comment]
Given that this works on nightly, the question is whether we want to push the fix to beta (and I guess identifying the fix). I'd be happy not to - doesn't seem like enough pain to warrant jumping trains. But I could be swayed.

**Comment by bluss** [comment]
It's unfortunate that the feature is only halfway introduced in Rust 1.1 in that case. I understand this is not clearly tracked for the release at all.

**Comment by eddyb** [comment]
@nrc if the fix is backported, then #26038 should probably be included with it, as it allows `Rc<RefCell<Unsized>>` and friends.

**Comment by arielb1** [comment]
Closed as 1.1 was released.
